django/db/migrations/serializer.py

```diff
@@ -244,8 +244,9 @@ def _format(self):
 
 class SetSerializer(BaseSequenceSerializer):
     def _format(self):
-        # Don't use the literal "{%s}" as it doesn't support empty set
-        return "set([%s])"
+        # When value is empty, the empty set should be serialized as "set()",
+        # not "{}" because {} is an empty dict.
+        return '{%s}' if self.value else 'set(%s)'
```

Comment:
In the projects I work on, migrations often contain more than auto-generated code. True, many start off as auto-generated code, but between schema tweaks and data migrations, there is plenty of written code. For this reason, my projects have success including `migrations` directories in flake8 runs.